### PR TITLE
Fixes #400 Kafka -> HBase pipeline

### LINKED ISSUE #400: Kafka -> HBase pipeline
We would like to see a use case where the source is kafka and a sliding window within the DAG calculates and updates an average and then writes this to hbase. 
Subtasks:
- [ ] #685 Pipeline: Fix runtime classpath and fat jar assembly to take advantage of jars on target node
- [ ] #686 Pipeline: Define all schema data using upickle
- [ ] #687 Pipeline: Create HBase tables, columns or assume they are precreated and return appropriate error info to AppMaster
- [ ] #688 Pipeline: Dynamically set zookeeper, kafka and hbase server locations in respective taskactor's prestart's
- [ ] #689 Pipeline: Possibly add CLI options to pipeline for topic, tables which would override pipeline.conf
- [ ] #690 Pipeline: Define DAG streaming behavior
- [ ] #714 Pipeline: extend KafkaConfig -> PipelineConfig
- [ ] #724 Pipeline: add data generator to pipeline example

### PR DESCRIPTION
**Pull request by kkasravi**
Some UT added (CpuProcessor, MemoryProcessor). Working on UT for KafkaProducer, CpuPersistor, MemoryPersistor, and HBaseConsumer.

[Review on Reviewable](https://reviewable.io/reviews/intel-hadoop/gearpump/740)

### CODE REVIEW

File: experiments/kafka-hbase-pipeline/src/test/scala/org/apache/gearpump/experiments/pipeline/ProcessorsSpec.scala [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    """
+      .stripMargin,
+    """
+      |{"id":"043ade58-2fbc-4fe2-8253-84ab181b8cfa","on":"2015-04-02T18:52:02.680078434Z","body":"{\"sample_id\":\"sample-0\",\"source_id\":\"src-4\",\"event_ts\":\"2015-04-02T18:52:04.78364878Z\",\"metrics\":[{\"dimension\":\"CPU\",\"metric\":\"total\",\"value\":27993996},{\"dimension\":\"CPU\",\"metric\":\"user\",\"value\":39017},{\"dimension\":\"CPU\",\"metric\":\"sys\",\"value\":23299},{\"dimension\":\"LOAD\",\"metric\":\"min\",\"value\":0},{\"dimension\":\"MEM\",\"metric\":\"free\",\"value\":15607009280},{\"dimension\":\"MEM\",\"metric\":\"used\",\"value\":163528704}]}"}
```
Comment (clockfly) [inline experiments/kafka-hbase-pipeline/src/test/scala/org/apache/gearpump/experiments/pipeline/ProcessorsSpec.scala]: Hi Kam,

you don't need to escape in """ """ pair

Comment (clockfly) [inline experiments/kafka-hbase-pipeline/src/test/scala/org/apache/gearpump/experiments/pipeline/ProcessorsSpec.scala]: I am sorry, looks like the backslash is added by github review system

**Comment by clockfly** [comment]
UT fails.

**Comment by kkasravi** [comment]
This uses the Reader Monad for dependency injection. In this case HBaseSink is injected into CpuPersistor and MemoryPersistor. This allows these classes to be tested by mocking HBaseSinkInterface and checking what parameters would be passed for HBaseSink.insert.

**Comment by clockfly** [comment]
+1

**Pull request merged**